src/Container/ServerRequestSwooleFactory.php

```diff
@@ -21,7 +28,35 @@ class ServerRequestSwooleFactory
     public function __invoke(ContainerInterface $container) : callable
     {
         return function (swoole_http_request $request) {
-            return ServerRequestFactory::fromSwoole($request);
+            // Aggregate values from Swoole request object
+            $get     = $request->get ?? [];
+            $post    = $request->post ?? [];
+            $cookie  = $request->cookie ?? [];
+            $files   = $request->files ?? [];
+            $server  = $request->server ?? [];
+            $headers = $request->header ?? [];
```

Comment:
We don't need anymore to convert `-` with `_` as reported in the previous code (as follows)?

```php
$headers = [];
if (isset($request->header)) {
    foreach ($request->header as $name => $value) {
        $headers[str_replace('-', '_', $name)] = $value;
    }
}
```

Reply:
No - we shouldn't do that. Headers are typically `-` separated, not `_` separated, and converting them to `_` will cause lookups to fail.

Reply:
Verified this behavior in both Swoole versions 2.2 and 4.0.1.